plugins/websocket_server/WebsocketServer.cc

```diff
@@ -239,9 +256,40 @@ bool WebsocketServer::Load(const tinyxml2::XMLElement *_elem)
   info.port = port;
   info.iface = NULL;
   info.protocols = &this->protocols[0];
-  // We are not using SSL right now
-  info.ssl_cert_filepath        = NULL;
-  info.ssl_private_key_filepath = NULL;
+
+  if (!sslCertFile.empty() && !sslPrivateKeyFile.empty())
+  {
+    // Fail if the certificate file cannot be opened.
+    if (!ignition::common::exists(sslCertFile))
+    {
+      ignerr << "SSL certificate file[" << sslCertFile
+        << "] does not exist. Quitting.\n";
+      return false;
+    }
+
+    // Fail if the private key file cannot be opened.
+    if (!ignition::common::exists(sslPrivateKeyFile))
+    {
+      ignerr << "SSL private key file[" << sslPrivateKeyFile
+        << "] does not exist. Quitting.\n";
+      return false;
+    }
+
+    // Store SSL configuration.
+    info.options = LWS_SERVER_OPTION_DO_SSL_GLOBAL_INIT;
+
+    info.ssl_cert_filepath = sslCertFile.c_str();
+    info.ssl_private_key_filepath = sslPrivateKeyFile.c_str();
+  }
+  else if (!sslCertFile.empty() || !sslPrivateKeyFile.empty())
```

Comment:
This should likely be
```suggestion
  else if (sslCertFile.empty() || sslPrivateKeyFile.empty())
```

Reply:
Fixed: d98a03e33048c3507988c6b250c44cf39a917dc4